addons/account/company.py

```diff
@@ -34,6 +34,20 @@ class ResCompany(models.Model):
     property_stock_account_input_categ = fields.Many2one('account.account')
     property_stock_account_output_categ = fields.Many2one('account.account')
     property_stock_valuation_account_id = fields.Many2one('account.account')
+    bank_journal_ids = fields.One2many('account.journal', 'company_id', domain=[('type', '=', 'bank'), ('bank_account_id', '!=', False)],
```

Comment:
remove this bank_journal_ids field, we don't need this. (company_id is set on journals)
You take in the footer all journal belonging to this company that have the display_on_footer boolean set